bundles/org.openhab.binding.etherrain/src/main/java/org/openhab/binding/etherrain/internal/api/EtherRainCommunication.java

```diff
@@ -0,0 +1,243 @@
+            return false;
+        }
+
+        return true;
+    }
+
+    public boolean commandLogout() {
+        try {
+            sendGet("/ergetcfg.cgi?m=o");
+        } catch (IOException e) {
+            logger.warn("Could not send logout command to etherrain: {}", e.getMessage());
+            return false;
+        }
+
+        return true;
+    }
+
+    private List<String> sendGet(String command) throws IOException {
+        String url = "http://" + address + ":" + port + "/" + command;
+
+        ContentResponse response;
+
+        try {
+            response = httpClient.newRequest(url).timeout(HTTP_TIMEOUT, TimeUnit.SECONDS).send();
+            if (response.getStatus() != HttpURLConnection.HTTP_OK) {
+                logger.warn("Etherrain return status other than HTTP_OK : {}", response.getStatus());
+                return new LinkedList<String>();
+            }
+        } catch (InterruptedException | TimeoutException | ExecutionException e) {
+            logger.warn("Could not connect to Etherrain with exception: {}", e.getMessage());
+            return new LinkedList<String>();
+        }
+
+        return new BufferedReader(new StringReader(response.getContentAsString())).lines().collect(Collectors.toList());
+    }
+
+    private static EtherRainUdpResponse updBroadcast() {
+        DatagramSocket c = null;
+
+        // Find the server using UDP broadcast
+        try {
+            // Open a random port to send the package
+            c = new DatagramSocket();
+            c.setSoTimeout(BROADCAST_TIMEOUT);
+            c.setBroadcast(true);
+
+            byte[] sendData = BROADCAST_DISCOVERY_MESSAGE.getBytes();
+
+            // Try the 255.255.255.255 first
+            DatagramPacket sendPacket = new DatagramPacket(sendData, sendData.length,
+                    InetAddress.getByName("255.255.255.255"), BROADCAST_DISCOVERY_PORT);
+            c.send(sendPacket);
+
+            // Wait for a response
+            byte[] recvBuf = new byte[15000];
+            DatagramPacket receivePacket;
+            try {
+                receivePacket = new DatagramPacket(recvBuf, recvBuf.length);
+                c.receive(receivePacket);
+            } catch (SocketTimeoutException e) {
+                return new EtherRainUdpResponse();
+            }
+
+            // Check if the message is correct
+            String message = new String(receivePacket.getData()).trim();
+
+            if (message.length() == 0) {
+                return new EtherRainUdpResponse();
+            }
+
+            String addressBC = receivePacket.getAddress().getHostAddress();
+            String deviceTypeBC = message.replaceAll(BROADCAST_RESPONSE_DISCOVER_PATTERN, "$1");
+            String unqiueNameBC = message.replaceAll(BROADCAST_RESPONSE_DISCOVER_PATTERN, "$2");
+            int portBC = Integer.parseInt(message.replaceAll(BROADCAST_RESPONSE_DISCOVER_PATTERN, "$3"));
```

Comment:
I'd much prefer you used a static `Pattern` instance that you created `Matcher` instances from and then use `matcher.group(..)` to get the group captured values instead.. It would be much more performant since you aren't forcing the regex compiler to constantly compile the same pattern over and over again.
But treat this as more a recommendation as an area of improvement and not as a required change for this PR.
```suggestion
            Matcher matcher = pattern.matcher(message);
            if(matcher.find()){
                String deviceTypeBC = matcher.group(1);
                String unqiueNameBC = matcher.group(2);
                int portBC = Integer.parseInt(matcher.group(3));
            }
```
The above assumes that you actually want the pattern to match, but here is a version that more closely reflects what you are currently doing while still being more performant.
```suggestion
            Matcher matcher = pattern.matcher(message);
            String deviceTypeBC = matcher.replaceAll("$1");
            String unqiueNameBC = matcher.replaceAll("$2");
            int portBC = Integer.parseInt(matcher.replaceAll("$3"));
```
Regex isn't cheap in java, so saving a Pattern instance is always worth it.

Reply:
No worries. I have implemented this. Will test and I don't see why it won't work.

Reply:
I went ahead and fixed this the way you requested.